core/object/object.h

```diff
@@ -47,103 +47,471 @@ template <typename T>
+	// all subclasses are included automatically.
+	// Editing the property will show a dialog for picking a node from the scene.
+	// This is similar to `PROPERTY_HINT_RESOURCE_TYPE`, but for `NodePath` properties.
+	// Use `PROPERTY_HINT_NODE_TYPE` for `Node` properties instead.
 	PROPERTY_HINT_NODE_PATH_VALID_TYPES,
-	PROPERTY_HINT_SAVE_FILE, ///< a file path must be passed, hint_text (optionally) is a filter "*.png,*.wav,*.doc,". This opens a save dialog
-	PROPERTY_HINT_GLOBAL_SAVE_FILE, ///< a file path must be passed, hint_text (optionally) is a filter "*.png,*.wav,*.doc,". This opens a save dialog
-	PROPERTY_HINT_INT_IS_OBJECTID, // Deprecated.
+
+	// Hints that a `String` property is a path to a file.
+	// Editing it will show a file dialog for picking the path for the file to be saved at.
+	// The dialog has access to the project's directory.
+	// The hint string can be a set of filters with wildcards like `"*.png,*.jpg"`. See also `PROPERTY_HINT_FILE`.
+	PROPERTY_HINT_SAVE_FILE,
+
+	// Hints that a `String` property is a path to a file.
+	// Editing it will show a file dialog for picking the path for the file to be saved at.
+	// The dialog has access to the entire filesystem.
+	// The hint string can be a set of filters with wildcards like `"*.png,*.jpg"`. See also `PROPERTY_HINT_GLOBAL_FILE`.
+	PROPERTY_HINT_GLOBAL_SAVE_FILE,
+
+	// Deprecated.
+	PROPERTY_HINT_INT_IS_OBJECTID,
+
+	// Hints that an `int` property is a pointer. Used by GDExtension.
 	PROPERTY_HINT_INT_IS_POINTER,
+
+	// Hints that a property is an [Array] with the stored type specified in the hint string.
 	PROPERTY_HINT_ARRAY_TYPE,
+
+	// Hints that a string property is a locale code.
+	// Editing it will show a locale dialog for picking language and country.
 	PROPERTY_HINT_LOCALE_ID,
+
+	// Hints that a dictionary property is a string translation map.
+	// Dictionary keys are locale codes; values are translated strings.
 	PROPERTY_HINT_LOCALIZABLE_STRING,
-	PROPERTY_HINT_NODE_TYPE, ///< a node object type
-	PROPERTY_HINT_HIDE_QUATERNION_EDIT, /// Only Node3D::transform should hide the quaternion editor.
+
+	// Hints that a property is an instance of a `Node`-derived type, optionally specified via the hint string (e.g. `"Node2D"`).
+	// The hint string can specify multiple classes separated by commas (e.g. `"Node2D,Node3D"`);
+	// all subclasses are included automatically.
+	// Editing the property will show a dialog for picking a node from the scene.
+	// This is similar to `PROPERTY_HINT_RESOURCE_TYPE`, but for `Node` properties.
+	// Use `PROPERTY_HINT_NODE_PATH_VALID_TYPES` for `NodePath` properties instead.
+	PROPERTY_HINT_NODE_TYPE,
+
+	// Hints that a `Quaternion` property should disable the temporary Euler editor.
+	PROPERTY_HINT_HIDE_QUATERNION_EDIT,
+
+	// Hints that a string property is a password, and every character is replaced
+	// with the secret character in the inspector.
 	PROPERTY_HINT_PASSWORD,
```

Comment:
It's also acting as `PROPERTY_HINT_PLACEHOLDER_TEXT` and use `hint_string` as placeholder.